server/src/controllers/federated_controller.ts

```diff
@@ -0,0 +1,246 @@
+
+import createDebug from "debug";
+import WebSocket from 'ws'
+import { v4 as randomUUID } from 'uuid'
+import msgpack from 'msgpack-lite'
+
+import type { WeightsContainer } from '@epfml/discojs'
+import {
+  aggregator as aggregators,
+  client,
+  serialization,
+} from '@epfml/discojs'
+
+import { TrainingController } from "./training_controller.js";
+
+import Messages = client.messages
+import MessageTypes = Messages.type
+import FederatedMessages = client.federated.messages
+
+const debug = createDebug("server:controllers:federated")
+
+// Arbitrarily set a minimum of two participants in a federated training
+const MIN_PARTICIPANTS = 2
+
+export class FederatedController extends TrainingController {
+  /**
+   * Aggregators for each hosted task.
+    By default the server waits for 100% of the nodes to send their contributions before aggregating the updates
+   */
+  #aggregator = new aggregators.MeanAggregator(undefined, 1, 'relative')
+  /**
+   * Promise containing the current round's results. To be awaited on when providing clients
+   * with the most recent result.
+   */
+  #result: Promise<WeightsContainer> | undefined = undefined
+  /**
+   * The last global model aggregated. The model is already serialized and 
+   * can be sent to participants joining mid-training or contributing to previous rounds
+   */
+  #latestGlobalModel: serialization.weights.Encoded | undefined = undefined
```

Comment:
"Model" makes me think that it also contains the architecture.

```suggestion
  #latestGlobalWeights: serialization.weights.Encoded | undefined = undefined
```